history-deleter/history.js

```diff
@@ -0,0 +1,58 @@
+// A useful way to extract the domain from a url.
+function get_hostname(url) {
+  var a = document.createElement('a');
+  a.href = url;
+  return a.hostname;
+}
+
+function no_history(hostname) {
+  document.getElementById('history').innerHTML = `No history for ${hostname}.`;
+}
+
+// When the page is loaded find the current tab and then use that to query
+// the history.
+chrome.tabs.query({active: true, currentWindow: true}, function(tabs) {
+  var list = document.getElementById('history');
+  var hostname = get_hostname(tabs[0].url);
+  chrome.history.search(
+    // Search for all history entries for the current windows domain.
+    {text: hostname},
```

Comment:
You can also pass `maxResults: 5` to limit the search to 5 results, rather than limiting to 5 in the loop below.